dbuf-lsp/src/navigation_handler/navigation/find_definition.rs

```diff
@@ -0,0 +1,177 @@
+                Some(TypeDefinition::Message(m)) => {
+                    return m
+                        .iter()
+                        .find(|f| f.name.as_ref() == field)
+                        .map(|f| f.name.get_location().to_lsp())
+                        .unwrap_or_else(|| {
+                            panic!("field not found\n{:#?}", symbol);
+                        })
+                        .into()
+                }
+                Some(TypeDefinition::Enum(branches)) => {
+                    for b in branches.iter() {
+                        for c in b.constructors.iter() {
+                            if c.name.as_ref() != constructor {
+                                continue;
+                            }
+                            return c
+                                .iter()
+                                .find(|f| f.name.as_ref() == field)
+                                .map(|f| f.name.get_location().to_lsp())
+                                .unwrap_or_else(|| {
+                                    panic!("field not found\n{:#?}", symbol);
+                                })
+                                .into();
+                        }
+                    }
+                }
+                None => {}
+            }
+            panic!("field not found\n{:#?}", symbol);
+        }
+        Symbol::Alias { t, branch_id, name } => {
+            let parsed = navigator.parsed;
+            let body = parsed
+                .iter()
+                .find(|d| d.name.as_ref() == t)
+                .map(|d| &d.data.body);
+
+            if let Some(TypeDefinition::Enum(e)) = body {
+                let b = e.get(*branch_id).unwrap_or_else(|| {
+                    panic!("alias not found\n{:#?}", symbol);
+                });
+
+                for p in b.patterns.iter() {
+                    let ans = find_alias_in_pattern(p, name);
+                    if ans.is_some() {
+                        return ans;
+                    }
+                }
+            }
+            panic!("alias not found\n{:#?}", symbol);
+        }
+        Symbol::Constructor(constructor) => {
+            let elaborated = navigator.elaborated;
+            let t = elaborated
+                .get_constructor_type(constructor)
+                .unwrap_or_else(|| {
+                    panic!("constructor not found\n{:#?}", symbol);
+                });
+
+            let declaration = navigator
+                .parsed
+                .iter()
+                .find(|d| d.name.as_ref() == t)
+                .unwrap_or_else(|| {
+                    panic!("constructor not found\n{:#?}", symbol);
+                });
+
+            match &declaration.body {
+                TypeDefinition::Message(_) => {
+                    panic!("symbol::constructor shouldn't be returned for messages");
+                }
+                TypeDefinition::Enum(branches) => {
+                    for b in branches.iter() {
```

Comment:
[`std::iter::Iterator::flat_map`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.flat_map) can be used to flatten nested structure